src/plugins/vis_type_timelion/server/series_functions/es/lib/create_date_agg.js

```diff
@@ -41,11 +41,12 @@ export default function createDateAgg(config, tlConfig, scriptedFields) {
         },
       };
     } else if (metric[0] && metric[1]) {
-      const metricName = metric[0] + '(' + metric[1] + ')';
+      const [field, percentArgs] = metric[1].split(/:(\d.*)/);
+      const metricName = metric[0] + '(' + field + ')';
       dateAgg.time_buckets.aggs[metricName] = {};
-      dateAgg.time_buckets.aggs[metricName][metric[0]] = buildAggBody(metric[1], scriptedFields);
-      if (metric[0] === 'percentiles' && metric[2]) {
-        let percentList = metric[2].split(',');
+      dateAgg.time_buckets.aggs[metricName][metric[0]] = buildAggBody(field, scriptedFields);
```

Comment:
what if we do it a little bit obvious. I see we are dealing with 3 cases. 

```
const [metricName, metricArgs] = metric.split(/:(.+)/);
   
if (metricName === 'count') {

} else if (metricName === 'percentiles') {

} else if (metricName, metricArgs){

}
```

Reply:
oh... sorry, I see we need 
```
dateAgg.time_buckets.aggs[metricName] = {};
dateAgg.time_buckets.aggs[metricName][metric[0]] = buildAggBody(field, scriptedFields);
``` 
for percentiles too.

Reply:
Now, it's cleaned up from array indexes at least